src/GafferScene/IECoreScenePreview/PrimitiveAlgo.cpp

```diff
@@ -784,8 +990,6 @@ IECoreScene::PrimitivePtr PrimitiveAlgo::mergePrimitives(
 				}
 			}
 
-			// For meshes and curves, the interpolation we need depends on the interpolations of all the variables
-			// being combined.
 			varInfo.interpolation = mergeInterpolations( resultTypeId, varInfo.interpolation, var.interpolation, name );
```

Comment:
Perhaps `mergeInterpolations` could be a static method on `ResultStruct`? Then the implementation of that function only needs to consider one primitive type, and the logic is simpler.